src/main/java/com/amazon/opendistro/elasticsearch/performanceanalyzer/reader/ReaderMetricsProcessor.java

```diff
@@ -201,8 +219,41 @@ public void trimOldSnapshots() throws Exception {
     trimMap(shardRqMetricsMap, RQ_SNAPSHOTS);
     trimMap(httpRqMetricsMap, HTTP_RQ_SNAPSHOTS);
     trimMap(masterEventMetricsMap, MASTER_EVENT_SNAPSHOTS);
-    trimDatabases(
-        metricsDBMap, MAX_DATABASES, PluginSettings.instance().shouldCleanupMetricsDBFiles());
+
+    boolean deleteDBFiles = PluginSettings.instance().shouldCleanupMetricsDBFiles();
+    if (metricsDBMap.size() > MAX_DATABASES) {
+      Map.Entry<Long, MetricsDB> lowestEntry = metricsDBMap.pollFirstEntry();
```

Comment:
by lowest do you mean oldest?

Reply:
Yes, will change the variable names to reflect this.